src/runtime-rs/crates/resource/src/volume/share_fs_volume.rs

```diff
@@ -60,10 +64,51 @@ impl ShareFsVolume {
                     }
                     Ok(src) => src,
                 };
+                let dest = [KATA_SHARED_SANDBOX_PATH, &file_name].join("/");
+
+                let src_metadata = std::fs::metadata(src.clone());
+                let cloned_src = src_metadata?.clone();
+
+                let mut file = File::open(&src)
+                    .with_context(|| format!("Failed to open file: {:?}", src))?;
+
+                let mut buffer = Vec::new();
+                file.read_to_end(&mut buffer)
+                    .with_context(|| format!("Failed to read file: {:?}", src))?;
+
+                let r = agent::CopyFileRequest {
+                    path: dest.clone(),
+                    file_size: cloned_src.len() as i64,
+                    uid: cloned_src.uid() as i32,
+                    gid: cloned_src.gid() as i32,
+                    file_mode: cloned_src.mode(),
+                    data: buffer,
+                    ..Default::default()
+                };
 
+                debug!(
+                    sl!(),
+                    "copy_file: {:?} to {:?}",
+                    &src,
+                    dest.clone()
+                );
+                agent
+                    .copy_file(r)
+                    .await
+                    .context("copy file request failed")?;
+
+                // This is where we set the value for the guest path
                 if src.is_file() {
```

Comment:
Thanks @ananos , this PR generally LGTM.  One more question, should the operation above be handled in this if statement?  If it is a none regular file, I think we don't need to copy this guest ?

Reply:
you're absolutely right! our bad -- should be now fixed.